internal/query/projection/project_relational.go

```diff
@@ -0,0 +1,175 @@
+package projection
+
+import (
+	"context"
+
+	repoDomain "github.com/zitadel/zitadel/backend/v3/domain"
+	"github.com/zitadel/zitadel/internal/eventstore"
+	"github.com/zitadel/zitadel/internal/eventstore/handler/v2"
+	"github.com/zitadel/zitadel/internal/repository/project"
+	"github.com/zitadel/zitadel/internal/zerrors"
+)
+
+const (
+	ProjectRelationalTable                       = "zitadel.projects"
+	ProjectRelationalShouldAssertRoleCol         = "should_assert_role"
+	ProjectRelationalIsAuthorizationRequiredCol  = "is_authorization_required"
+	ProjectRelationalIsProjectAccessRequiredCol  = "is_project_access_required"
+	ProjectRelationalUsedLabelingSettingOwnerCol = "used_labeling_setting_owner"
+)
+
+type projectRelationalProjection struct{}
+
+func (*projectRelationalProjection) Name() string {
+	return ProjectRelationalTable
+}
+
+func newProjectRelationalProjection(ctx context.Context, config handler.Config) *handler.Handler {
+	return handler.NewHandler(ctx, &config, new(projectRelationalProjection))
+}
+
+func (p *projectRelationalProjection) Reducers() []handler.AggregateReducer {
+	return []handler.AggregateReducer{
+		{
+			Aggregate: project.AggregateType,
+			EventReducers: []handler.EventReducer{
+				{
+					Event:  project.ProjectAddedType,
+					Reduce: p.reduceProjectAdded,
+				},
+				{
+					Event:  project.ProjectChangedType,
+					Reduce: p.reduceProjectChanged,
+				},
+				{
+					Event:  project.ProjectDeactivatedType,
+					Reduce: p.reduceProjectDeactivated,
+				},
+				{
+					Event:  project.ProjectReactivatedType,
+					Reduce: p.reduceProjectReactivated,
+				},
+				{
+					Event:  project.ProjectRemovedType,
+					Reduce: p.reduceProjectRemoved,
+				},
+			},
+		},
+	}
+}
+
+func (p *projectRelationalProjection) reduceProjectAdded(event eventstore.Event) (*handler.Statement, error) {
+	e, ok := event.(*project.ProjectAddedEvent)
+	if !ok {
+		return nil, zerrors.ThrowInternalf(nil, "HANDL-Oox5e", "reduce.wrong.event.type %s", project.ProjectAddedType)
+	}
+
+	return handler.NewCreateStatement(
+		e,
+		[]handler.Column{
+			handler.NewCol(InstanceID, e.Aggregate().InstanceID),
+			handler.NewCol(OrganizationID, e.Aggregate().ResourceOwner),
+			handler.NewCol(ID, e.Aggregate().ID),
+			handler.NewCol(CreatedAt, e.CreationDate()),
+			handler.NewCol(UpdatedAt, e.CreationDate()),
+			handler.NewCol(ProjectColumnName, e.Name),
+			handler.NewCol(State, repoDomain.ProjectStateActive),
+			handler.NewCol(ProjectRelationalShouldAssertRoleCol, e.ProjectRoleAssertion),
+			handler.NewCol(ProjectRelationalIsAuthorizationRequiredCol, e.ProjectRoleCheck),
+			handler.NewCol(ProjectRelationalIsProjectAccessRequiredCol, e.HasProjectCheck),
+			handler.NewCol(ProjectRelationalUsedLabelingSettingOwnerCol, e.PrivateLabelingSetting),
```

Comment:
The event field names are very confusing and I'm not sure if I got this mapping right. @livio-a can you please have a look at this?

Reply:
LGTM and matches the naming of the project v2 api pretty much: https://github.com/zitadel/zitadel/blob/main/proto/zitadel/project/v2beta/project_service.proto#L700-L707